src/Network/Network/NetworkVirtualAppliance/NetworkProfile/NewVirtualApplianceNetworkProfileCommand.cs

```diff
@@ -0,0 +1,114 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+using System.Linq;
+using Microsoft.Azure.Commands.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualApplianceNetworkProfile",
+        SupportsShouldProcess = true),
+        OutputType(typeof(PSVirtualApplianceNetworkProfile))]
+    public class NewVirtualApplianceNetworkProfileCommand : VirtualApplianceNetworkProfileBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = false,
+            HelpMessage = "The network interface configurations of the network profile.")]
+        [ValidateNotNullOrEmpty]
+        public PSVirtualApplianceNetworkInterfaceConfiguration[] NetworkInterfaceConfigurations { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var networkProfile = new PSVirtualApplianceNetworkProfile
+            {
+                NetworkInterfaceConfigurations = this.NetworkInterfaceConfigurations.ToList()
+            };
+
+            WriteObject(networkProfile);
+        }
+    }
+
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualApplianceNetworkInterfaceConfiguration",
+        SupportsShouldProcess = true),
+        OutputType(typeof(PSVirtualApplianceNetworkInterfaceConfiguration))]
+    public class NewVirtualApplianceNetworkInterfaceConfigurationCommand : VirtualApplianceNetworkProfileBaseCmdlet
```

Comment:
Please put only one cmdlet in one csharp file. It may confuse others for reading it.